server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalAutoDateHistogram.java

```diff
@@ -500,15 +500,24 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
         BucketReduceResult reducedBucketsResult = reduceBuckets(aggregations, reduceContext);
 
         if (reduceContext.isFinalReduce()) {
+            // Because auto-date-histo can perform multiple reductions while merging buckets, we need to pretend this is
+            // not the final reduction to prevent pipeline aggs from creating their result early.  However we want
+            // to reuse the multiBucketConsumer so that max_buckets breaker is correctly accounted for
+            ReduceContext fakeReduceContext = new ReduceContext(reduceContext.bigArrays(), reduceContext.scriptService(),
```

Comment:
I wonder if `penultimateReduceContext` or `stepBeforeFinalReduceContext` is clearer? on first glance, the term `fake` could read like it's a mistake something, I feel it implies something like "mocked reduce context" but that's not what's going on here. thoughts?

Reply:
++ you're right, that name could be confusing to future viewers.  Will give it something more descriptive (I rather like "penultimate"!)